File: app/src/main/kotlin/com/apex/trader/presentation/screen/signal/SignalDetailScreen.kt
package com.apex.trader.presentation.screen.signal

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material3.Button
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.apex.trader.domain.strategy.Signal
import com.apex.trader.domain.strategy.SignalReason
import com.apex.trader.domain.strategy.SignalSide
import com.apex.trader.presentation.component.ApexCard
import com.apex.trader.presentation.component.CandleChart
import com.apex.trader.presentation.component.ConfidenceBadge
import com.apex.trader.presentation.component.KeyValueRow
import com.apex.trader.presentation.component.SidePill
import com.apex.trader.presentation.theme.ApexBear
import com.apex.trader.presentation.theme.ApexBull
import com.apex.trader.presentation.theme.ApexHighlight
import com.apex.trader.presentation.theme.ApexTextMuted

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun SignalDetailScreen(
    symbol: String,
    onTradeClick: () -> Unit,
    onBack: () -> Unit,
    vm: SignalDetailViewModel = hiltViewModel()
) {
    val ui by vm.ui.collectAsStateWithLifecycle()

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text(symbol) },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, "Back")
                    }
                },
                actions = {
                    IconButton(onClick = { vm.refresh() }) {
                        Icon(Icons.Default.Refresh, "Refresh")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.background
                )
            )
        }
    ) { padding ->
        Column(
            modifier = Modifier.fillMaxSize().padding(padding),
            verticalArrangement = Arrangement.spacedBy(0.dp)
        ) {
            when {
                ui.loading -> Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                    CircularProgressIndicator()
                }
                ui.error != null -> Text(
                    text = "⚠ ${ui.error}",
                    color = MaterialTheme.colorScheme.error,
                    modifier = Modifier.padding(16.dp)
                )
                else -> Content(ui, onTradeClick)
            }
        }
    }
}

@Composable
private fun Content(ui: SignalDetailUiState, onTradeClick: () -> Unit) {
    val sig = ui.signal
    LazyColumn(
        modifier = Modifier.fillMaxSize(),
        verticalArrangement = Arrangement.spacedBy(10.dp),
        contentPadding = PaddingValues(12.dp)
    ) {
        item {
            ApexCard {
                Column {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Text(ui.symbol, style = MaterialTheme.typography.headlineMedium)
                        Spacer(Modifier.padding(end = 8.dp))
                        if (sig != null) SidePill(if (sig.side == SignalSide.LONG) "LONG" else "SHORT")
                        Spacer(Modifier.weight(1f))
                        if (sig != null) ConfidenceBadge(sig.confidence)
                    }
                    if (sig == null) {
                        Spacer(Modifier.height(6.dp))
                        Text(
                            "No active high-confidence setup right now. The chart shows the latest 200 candles for context.",
                            color = ApexTextMuted,
                            style = MaterialTheme.typography.bodyMedium
                        )
                    }
                }
            }
        }

        item {
            ApexCard {
                CandleChart(candles = ui.candles, signal = sig)
            }
        }

        if (sig != null) {
            item {
                ApexCard {
                    Column {
                        Text("Trade plan", style = MaterialTheme.typography.titleMedium)
                        Spacer(Modifier.height(8.dp))
                        KeyValueRow("Entry", "%.6f".format(sig.plan.entry))
                        KeyValueRow("Stop loss", "%.6f".format(sig.plan.stopLoss), valueColor = ApexBear)
                        KeyValueRow("TP1 (${"%.1f".format(sig.plan.riskRewardR1)}R)", "%.6f".format(sig.plan.takeProfit1), valueColor = ApexBull)
                        KeyValueRow("TP2 (${"%.1f".format(sig.plan.riskRewardR2)}R)", "%.6f".format(sig.plan.takeProfit2), valueColor = ApexBull)
                        KeyValueRow("TP3 (${"%.1f".format(sig.plan.riskRewardR3)}R)", "%.6f".format(sig.plan.takeProfit3), valueColor = ApexBull)
                        KeyValueRow("ATR", "%.6f".format(sig.plan.atr))
                        KeyValueRow("ADX", "%.1f".format(sig.adx))
                        KeyValueRow("RSI", "%.1f".format(sig.rsi))
                        KeyValueRow("Volume surge", "%.2fx".format(sig.volumeSurge))
                    }
                }
            }
            item {
                ApexCard {
                    Column {
                        Text("Why this signal", style = MaterialTheme.typography.titleMedium)
                        Spacer(Modifier.height(6.dp))
                        sig.reasons.forEach { ReasonRow(it) }
                    }
                }
            }
            item {
                Button(
                    onClick = onTradeClick,
                    modifier = Modifier.fillMaxWidth()
                ) { Text("Open trade") }
            }
        }
    }
}

@Composable
private fun ReasonRow(r: SignalReason) {
    Row(
        modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Text(
            text = if (r.passed) "✓" else "✗",
            color = if (r.passed) ApexBull else ApexBear,
            modifier = Modifier.padding(end = 8.dp),
            style = MaterialTheme.typography.titleMedium
        )
        Column(modifier = Modifier.weight(1f)) {
            Text(r.label, style = MaterialTheme.typography.bodyMedium)
            Text(r.detail, color = ApexTextMuted, style = MaterialTheme.typography.labelSmall)
        }
        Text(
            "+${r.weight.toInt()}",
            color = ApexHighlight,
            style = MaterialTheme.typography.labelMedium
        )
    }
}

File: lib/features/signal/signal_detail_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../core/theme.dart';
import '../../data/models/candle.dart';
import '../../data/models/timeframe.dart';
import '../../domain/strategy.dart';
import '../../providers.dart';
import '../../widgets/candle_chart.dart';
import '../../widgets/common.dart';

class SignalDetailScreen extends ConsumerStatefulWidget {
  const SignalDetailScreen({super.key, required this.symbol, required this.onTrade});
  final String symbol;
  final VoidCallback onTrade;

  @override
  ConsumerState<SignalDetailScreen> createState() => _SignalDetailScreenState();
}

class _SignalDetailScreenState extends ConsumerState<SignalDetailScreen> {
  bool _loading = true;
  String? _error;
  List<Candle> _candles = const [];
  Signal? _signal;

  @override
  void initState() {
    super.initState();
    _refresh();
  }

  Future<void> _refresh() async {
    setState(() {
      _loading = true;
      _error = null;
    });
    try {
      final settings = await ref.read(settingsRepoProvider).load();
      final api = ref.read(binanceApiProvider);
      final ltf = Timeframe.fromCode(settings.ltfTimeframe);
      final candles = await api.getCandles(widget.symbol, ltf, limit: 200);
      final signal = await ref.read(scannerProvider).evaluateOne(widget.symbol, settings);
      if (!mounted) return;
      setState(() {
        _candles = candles;
        _signal = signal;
        _loading = false;
      });
    } catch (e) {
      if (!mounted) return;
      setState(() {
        _loading = false;
        _error = e.toString();
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.symbol),
        actions: [
          IconButton(onPressed: _refresh, icon: const Icon(Icons.refresh)),
        ],
      ),
      body: _loading
          ? const Center(child: CircularProgressIndicator())
          : _error != null
              ? Padding(
                  padding: const EdgeInsets.all(16),
                  child: Text('⚠ $_error', style: const TextStyle(color: ApexColors.bear)),
                )
              : _content(context),
    );
  }

  Widget _content(BuildContext context) {
    final s = _signal;
    return ListView(
      padding: const EdgeInsets.all(12),
      children: [
        ApexCard(
          child: Row(
            children: [
              Text(widget.symbol, style: Theme.of(context).textTheme.headlineSmall),
              const SizedBox(width: 10),
              if (s != null) SidePill(side: s.side == SignalSide.long ? 'LONG' : 'SHORT'),
              const Spacer(),
              if (s != null) ConfidenceBadge(score: s.confidence),
            ],
          ),
        ),
        const SizedBox(height: 10),
        ApexCard(child: CandleChart(candles: _candles, signal: s)),
        const SizedBox(height: 10),
        if (s != null) ...[
          ApexCard(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text('Trade plan', style: Theme.of(context).textTheme.titleMedium),
                const SizedBox(height: 6),
                KeyValueRow(label: 'Entry', value: s.plan.entry.toStringAsFixed(6)),
                KeyValueRow(
                    label: 'Stop loss',
                    value: s.plan.stopLoss.toStringAsFixed(6),
                    valueColor: ApexColors.bear),
                // Trend strategies exit on an indicator (EMA cross), not
                // fixed targets — they emit 0 TPs. Show that instead of 0.0.
                if (s.plan.takeProfit1 <= 0 &&
                    s.plan.takeProfit2 <= 0 &&
                    s.plan.takeProfit3 <= 0)
                  const KeyValueRow(
                      label: 'Exit', value: 'EMA cross-back (no fixed TP)')
                else ...[
                  KeyValueRow(
                      label: 'TP1 (${s.plan.riskRewardR1}R)',
                      value: s.plan.takeProfit1.toStringAsFixed(6),
                      valueColor: ApexColors.bull),
                  KeyValueRow(
                      label: 'TP2 (${s.plan.riskRewardR2}R)',
                      value: s.plan.takeProfit2.toStringAsFixed(6),
                      valueColor: ApexColors.bull),
                  KeyValueRow(
                      label: 'TP3 (${s.plan.riskRewardR3}R)',
                      value: s.plan.takeProfit3.toStringAsFixed(6),
                      valueColor: ApexColors.bull),
                ],
                KeyValueRow(label: 'ATR', value: s.plan.atr.toStringAsFixed(6)),
                KeyValueRow(label: 'ADX', value: s.adx.toStringAsFixed(1)),
                KeyValueRow(label: 'RSI', value: s.rsi.toStringAsFixed(1)),
                KeyValueRow(label: 'Volume surge', value: '${s.volumeSurge.toStringAsFixed(2)}x'),
              ],
            ),
          ),
          const SizedBox(height: 10),
          ApexCard(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text('Why this signal', style: Theme.of(context).textTheme.titleMedium),
                const SizedBox(height: 6),
                ...s.reasons.map(_ReasonRow.new),
              ],
            ),
          ),
          const SizedBox(height: 10),
          SizedBox(
            width: double.infinity,
            child: ElevatedButton(onPressed: widget.onTrade, child: const Text('Open trade')),
          ),
        ] else
          const ApexCard(
            child: Text(
              'No active high-confidence setup right now. The chart shows the latest candles for context.',
              style: TextStyle(color: ApexColors.textMuted),
            ),
          ),
      ],
    );
  }
}

class _ReasonRow extends StatelessWidget {
  const _ReasonRow(this.r);
  final SignalReason r;

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 4),
      child: Row(
        crossAxisAlignment: CrossAxisAlignment.center,
        children: [
          Text(
            r.passed ? '✓' : '✗',
            style: TextStyle(
              color: r.passed ? ApexColors.bull : ApexColors.bear,
              fontSize: 18,
              fontWeight: FontWeight.w700,
            ),
          ),
          const SizedBox(width: 10),
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(r.label),
                Text(r.detail,
                    style: const TextStyle(color: ApexColors.textMuted, fontSize: 11)),
              ],
            ),
          ),
          Text('+${r.weight.toInt()}',
              style: const TextStyle(color: ApexColors.highlight, fontFamily: 'monospace')),
        ],
      ),
    );
  }
}
